### PR TITLE
azure: allow installer to consume `AZURE_AUTH_LOCATION` env for credentials file

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Installer uses `NewAuthorizerFromFileWithResource` [here][1], which uses `GetSettingsFromFile` [here][2] to locate and load the file with auth credentials.
`GetSettingsFromFile` [here][2] uses the `AZURE_AUTH_LOCATION` env [here][3] to locate the file with no way to override or specify explicitly.

Currently the installer uses the hard-coded location `~/.azure/osServicePrincipal.json` to load the credentials. But for CI, it would be important to override this
location to another location like we do for AWS [here][4]. So this change allows users to set `AZURE_AUTH_LOCATION` env to provider installer custom location to auth file.

[1]: https://github.com/Azure/go-autorest/blob/v12.0.0/autorest/azure/auth/auth.go#L243
[2]: https://github.com/Azure/go-autorest/blob/v12.0.0/autorest/azure/auth/auth.go#L287
[3]: https://github.com/Azure/go-autorest/blob/v12.0.0/autorest/azure/auth/auth.go#L289
[4]: https://github.com/openshift/release/blob/6c0b409639d6dcd074238e5396cddcc5c4da1510/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L373-L374

/cc @openshift/installer

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
:P the installer github team has incomplete list.

/cc @jstuever @wking

**Comment by abhinavdahiya** [comment]
Need this PR to get CI rolling, 
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3877/rehearse-3877-pull-ci-openshift-installer-master-e2e-azure/4#operator-run-template-e2e-azure---e2e-azure-container-setup

```console
level=fatal msg="failed to fetch Terraform Variables: failed to fetch dependency of \"Terraform Variables\": failed to fetch dependency of \"Bootstrap Ignition Config\": failed to fetch dependency of \"Master Machines\": failed to generate asset \"Platform Credentials Check\": creating Azure session: failed to retrieve credentials from user: EOF"
�[0G�[2K�[1;92m? �[1;99mazure subscription id [? for help] �[?25l�7�[999;999f�[6n�[6n
```

ping @jstuever @wking

File: pkg/asset/installconfig/azure/session.go [hunk 1]
```diff
@@ -16,15 +16,15 @@ import (
 
 const azureAuthEnv = "AZURE_AUTH_LOCATION"
 
-var authFilePath = os.Getenv("HOME") + "/.azure/osServicePrincipal.json"
+var defaultAuthFilePath = os.Getenv("HOME") + "/.azure/osServicePrincipal.json"
```
Comment (wking) [inline pkg/asset/installconfig/azure/session.go]: I prefer `filePath.Join` so we aren't so far off when we get around to compiling for Windows.  But whatever, the `+` concat approach is what we had before this PR.

File: pkg/asset/installconfig/azure/session.go [hunk 2]
```diff
@@ -35,18 +35,25 @@ type Credentials struct {
 // GetSession returns an azure session by using credentials found in ~/.azure/osServicePrincipal.json
 // and, if no creds are found, asks for them and stores them on disk in a config file
 func GetSession() (*Session, error) {
-	os.Setenv(azureAuthEnv, authFilePath)
-	return newSessionFromFile()
+	authFile := defaultAuthFilePath
+	if f, ok := os.LookupEnv(azureAuthEnv); ok && len(f) > 0 {
```
Comment (wking) [inline pkg/asset/installconfig/azure/session.go]: If you're going to gate on length anyway, you might as well just use [`os.Getenv`][1].

[1]: https://golang.org/pkg/os/#Getenv

File: pkg/asset/installconfig/azure/session.go (hunk 1 above)
Comment (abhinavdahiya) [reply pkg/asset/installconfig/azure/session.go]: fixed with https://github.com/openshift/installer/compare/4bff9ab1f5f879ccdabd9fa5730d21b711aec476...0e9453e723a95ec3a6290e2985d7926f8a3f7618

File: pkg/asset/installconfig/azure/session.go (hunk 2 above)
Comment (abhinavdahiya) [reply pkg/asset/installconfig/azure/session.go]: fixed with https://github.com/openshift/installer/compare/4bff9ab1f5f879ccdabd9fa5730d21b711aec476...0e9453e723a95ec3a6290e2985d7926f8a3f7618

**Pull request merged**